PantheonCMD/pcbuild.py

```diff
@@ -124,40 +111,150 @@ def coalesce_document(main_file, attributes=None, depth=0, top_level=True):
 
     # Open the file, iterate over lines
     if os.path.exists(main_file):
+
         with open(main_file) as input_file:
+
             # Iterate over content
             for line in input_file:
-                # Process comment block start and finish
-                if line.strip().startswith('////'):
+
+                # CONDITIONS
+
+                # Line matches the end of a condition_block
+                matches = re.match(r'^endif::(.*?)\[\]', line)
+
+                if matches:
+                    if matches.group(1) == '':
+                        condition_block = False
+                    else:
+                        if matches.group(1) == conditions_set:
+                            conditions_set = ''
+                            condition_block = False
+                    continue
+
+                # Line matches the middle of a condition block
+                if condition_block:
+                    continue
+
+                # Line matches the start of an ifdef condition
+                matches = re.match(r'^ifdef::(\S+)\[(.*?)\]', line)
+
+                if matches:
+                    conditions_missing = False
+                    conditions = matches.group(1)
+                    conditions_set = matches.group(1)
+
+                    # Multiple conditions - single match is enough
+                    if conditions.__contains__('+'):
+                        conditions_list = conditions.split('+')
+                        for condition in conditions_list:
+                            if not condition in attributes.keys():
+                                conditions_missing = True
+                                break
+
+                    # Multiple conditions - all must match
+                    elif conditions.__contains__(','):
+                        conditions_missing = False
+                        conditions_list = conditions.split(',')
+                        for condition in conditions_list:
+                            if not condition in attributes.keys():
+                                conditions_missing = True
+                            else:
+                                conditions_missing = True
+                        if conditions_missing:
+                            conditions_missing = False
```

Comment:
same here, both if and else statements result in `conditions_missing = True`. can't it be:

```suggestion
                        if conditions_missing:
                            conditions_missing = False
                        else:
                            conditions_missing = True        
```

Reply:
Hmm - yep, good point.

I'll run through this one again to see if there was something I was trying to do there. Maybe I've just confused myself.